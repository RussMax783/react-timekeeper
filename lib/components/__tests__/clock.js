'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

var _Clock = require('../Clock');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('Clock component', function () {
	var hour = 4;
	var minute = 42;

	var changeHour = jest.fn();
	var changeMinute = jest.fn();

	describe('renders correctly', function () {
		test('when unit is hour', function () {
			var unit = 'hour';
			var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_Clock.Clock, {
				config: {},
				unit: unit,
				hour: hour,
				minute: minute,
				changeHour: changeHour,
				changeMinute: changeMinute
			})).toJSON();
			expect(tree).toMatchSnapshot();
		});
		test('when unit is minute', function () {
			var unit = 'minute';
			var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_Clock.Clock, {
				config: {},
				unit: unit,
				hour: hour,
				minute: minute,
				changeHour: changeHour,
				changeMinute: changeMinute
			})).toJSON();
			expect(tree).toMatchSnapshot();
		});
	});
});