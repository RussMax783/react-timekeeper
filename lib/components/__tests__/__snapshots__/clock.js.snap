exports[`Clock component renders correctly when unit is hour 1`] = `
<div
  className="react-timekeeper__clock"
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "background": undefined,
      "borderRadius": "200px",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "220px",
      "position": "relative",
      "width": "220px",
    }
  }>
  <div
    className="react-timekeeper__clock-animations-wrapper">
    <div
      className="react-timekeeper__clock-animations"
      style={
        Object {
          "position": "absolute",
        }
      }>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 136.99999999999997,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 16.789764466969388,
            "width": 34,
            "zIndex": 5,
          }
        }>
        1
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 169.21023553303058,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 48.99999999999996,
            "width": 34,
            "zIndex": 5,
          }
        }>
        2
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 181,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 92.99999999999999,
            "width": 34,
            "zIndex": 5,
          }
        }>
        3
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 169.21023553303058,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 137,
            "width": 34,
            "zIndex": 5,
          }
        }>
        4
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 137.00000000000006,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 169.21023553303058,
            "width": 34,
            "zIndex": 5,
          }
        }>
        5
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 93.00000000000003,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 181,
            "width": 34,
            "zIndex": 5,
          }
        }>
        6
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 49,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 169.21023553303058,
            "width": 34,
            "zIndex": 5,
          }
        }>
        7
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 16.789764466969416,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 137.00000000000006,
            "width": 34,
            "zIndex": 5,
          }
        }>
        8
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 5,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 93.00000000000003,
            "width": 34,
            "zIndex": 5,
          }
        }>
        9
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 16.78976446696936,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 49.00000000000007,
            "width": 34,
            "zIndex": 5,
          }
        }>
        10
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 49.00000000000003,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 16.789764466969388,
            "width": 34,
            "zIndex": 5,
          }
        }>
        11
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 92.99999999999997,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 5,
            "width": 34,
            "zIndex": 5,
          }
        }>
        12
      </span>
      <svg
        className="react-timekeeper__clock-svgs"
        height={220}
        style={
          Object {
            "opacity": 1,
            "position": "relative",
          }
        }
        viewBox="0 0 220 220"
        width={220}
        xmlns="http://www.w3.org/2000/svg">
        <g
          transform="rotate(120 110 110)">
          <line
            stroke={undefined}
            strokeWidth="1"
            x1={110}
            x2={110}
            y1={110}
            y2={20} />
          <circle
            cx={110}
            cy={110}
            fill={undefined}
            r={1.5} />
          <circle
            cx={110}
            cy={22}
            fill={undefined}
            r={17} />
        </g>
      </svg>
    </div>
  </div>
</div>
`;

exports[`Clock component renders correctly when unit is minute 1`] = `
<div
  className="react-timekeeper__clock"
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "background": undefined,
      "borderRadius": "200px",
      "cursor": "pointer",
      "display": "inline-block",
      "height": "220px",
      "position": "relative",
      "width": "220px",
    }
  }>
  <div
    className="react-timekeeper__clock-animations-wrapper">
    <div
      className="react-timekeeper__clock-animations"
      style={
        Object {
          "position": "absolute",
        }
      }>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 136.99999999999997,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 16.789764466969388,
            "width": 34,
            "zIndex": 5,
          }
        }>
        05
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 169.21023553303058,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 48.99999999999996,
            "width": 34,
            "zIndex": 5,
          }
        }>
        10
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 181,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 92.99999999999999,
            "width": 34,
            "zIndex": 5,
          }
        }>
        15
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 169.21023553303058,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 137,
            "width": 34,
            "zIndex": 5,
          }
        }>
        20
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 137.00000000000006,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 169.21023553303058,
            "width": 34,
            "zIndex": 5,
          }
        }>
        25
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 93.00000000000003,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 181,
            "width": 34,
            "zIndex": 5,
          }
        }>
        30
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 49,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 169.21023553303058,
            "width": 34,
            "zIndex": 5,
          }
        }>
        35
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 16.789764466969416,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 137.00000000000006,
            "width": 34,
            "zIndex": 5,
          }
        }>
        40
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 5,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 93.00000000000003,
            "width": 34,
            "zIndex": 5,
          }
        }>
        45
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 16.78976446696936,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 49.00000000000007,
            "width": 34,
            "zIndex": 5,
          }
        }>
        50
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 49.00000000000003,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 16.789764466969388,
            "width": 34,
            "zIndex": 5,
          }
        }>
        55
      </span>
      <span
        style={
          Object {
            "borderRadius": "99px",
            "color": undefined,
            "display": "inline-block",
            "fontSize": "16px",
            "height": 34,
            "left": 92.99999999999997,
            "lineHeight": "34px",
            "opacity": 1,
            "pointerEvents": "none",
            "position": "absolute",
            "textAlign": "center",
            "top": 5,
            "width": 34,
            "zIndex": 5,
          }
        }>
        00
      </span>
      <svg
        className="react-timekeeper__clock-svgs"
        height={220}
        style={
          Object {
            "opacity": 1,
            "position": "relative",
          }
        }
        viewBox="0 0 220 220"
        width={220}
        xmlns="http://www.w3.org/2000/svg">
        <g
          transform="rotate(252 110 110)">
          <line
            stroke={undefined}
            strokeWidth="1"
            x1={110}
            x2={110}
            y1={110}
            y2={20} />
          <circle
            cx={110}
            cy={110}
            fill={undefined}
            r={1.5} />
          <circle
            cx={110}
            cy={22}
            fill={undefined}
            r={17} />
          <circle
            cx={110}
            cy={22}
            fill={undefined}
            r={4} />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
